python/engine/CMakeLists.txt

```diff
@@ -44,4 +44,25 @@ target_compile_options(pythonengine PRIVATE
 
 target_compile_definitions(pythonengine PRIVATE openstudio_scriptengine_EXPORTS SHARED_OS_LIBS)
 
-install(TARGETS pythonengine EXPORT openstudio DESTINATION bin COMPONENT "CLI")
+install(TARGETS pythonengine EXPORT openstudio DESTINATION ${LIB_DESTINATION_DIR} COMPONENT "CLI")
+
+# it goes into lib/ and we want to find:
+# ../EnergyPlus/libpython3.8.dylib
+if(APPLE)
+  set_target_properties(pythonengine PROPERTIES
+    INSTALL_RPATH "@loader_path;@loader_path/../EnergyPlus/"
+  )
+
+  install(CODE [[
+    execute_process(COMMAND "install_name_tool" -change "$<TARGET_FILE:Python::Python>" "@rpath/$<TARGET_FILE_NAME:Python::Python>" "${CMAKE_INSTALL_PREFIX}/lib/$<TARGET_FILE_NAME:pythonengine>")
+  ]])
```

Comment:
This is missing COMPONENT "CLI", will add after windows is done building.

On mac, we have no choice but to call install_name_tool.

The CMAKE_INSTALL_PREFIX is needed because if I use `$<TARGET_FILE:pythonengine>` it actually fixes the one in build/Products/ and not the one in the cpack staging area.